setup.py

```diff
@@ -11,14 +11,36 @@
 with open("HISTORY.rst") as fp:
     history = fp.read().replace(".. :changelog:", "")
 
-with open("requirements.txt") as fp:
-    requirements = list(filter(bool, (line.strip() for line in fp)))
-
-with open("requirements-dev.txt") as fp:
-    dev_requirements = list(filter(bool, (line.strip() for line in fp)))
+extras_require = {
+    "test": [
+        "build>=0.7",
+        "cython>=0.25.1",
+        'importlib-metadata;python_version<"3.8"',
+        "pytest>=6.0.0",
+        "pytest-mock>=1.10.4",
+        "pytest-virtualenv>=1.2.5",
+        "requests",
+        "virtualenv",
+    ],
+    "doctest": [
+        "ubelt >= 0.8.2",
+        "xdoctest>=0.10.0",
+    ],
+    "cov": [
+        "codecov>=2.0.5",
+        "coverage>=4.2",
+        "pytest-cov>=2.7.1",
+    ],
+    "docs": [
+        "pygments",
+        "sphinx>=4",
+        "sphinx-issues",
+        "sphinx-rtd-theme>=1.0",
+        "sphinxcontrib-moderncmakedomain>=3.19",
+    ],
+}
 
-with open("requirements-docs.txt") as fp:
-    doc_requirements = list(filter(bool, (line.strip() for line in fp)))
+extras_require["dev"] = extras_require["test"] + extras_require["cov"] + extras_require["doctest"]
```

Comment:
I'm leaving the dev requirement for now, though static dep lists make it harder to do these sort of combined requirements, so I'm focusing on listing the separate ones everywhere.

Reply:
Oh, actually, this was called "tests" or "requirements-dev" before, so maybe it would be best to just not add it at all, since we might want to remove it when moving to static configuration later.

Reply:
```suggestion
```